snowplow-tracker/src/main/java/com/snowplowanalytics/snowplow/internal/emitter/storage/EventStoreHelper.java

```diff
@@ -48,7 +55,35 @@
             "dateCreated TIMESTAMP DEFAULT CURRENT_TIMESTAMP)";
 
     // Prevents multiple instances being created and avoids memory leaks.
-    private static EventStoreHelper sInstance;
+    private static HashMap<String, EventStoreHelper> instances = new HashMap<>();
+
+    @NonNull
+    public synchronized static List<String> removeUnsentEventsExceptForNamespaces(@NonNull Context context, @Nullable List<String> allowedNamespaces) {
```

Comment:
Whats the use case for passing in `List<String> allowedNamespaces`?

Reply:
This method and the way is implemented is something I dislike. The problem is that the app can have zombie events inside EventStores associated to namespace no longer used. In that case the developer at startup can instrument the tracker(s) and delete all the EventStores EXCEPT the ones associated to the "allowedNamespaces", which are the namespaces of the just instrumented trackers.

I think you suggested the right way to go, allowing the zombie events to be associated to emitter configurations and automatically flush them. But it was a too big feature too implement now.